pkg/api/vlabs/validate.go

```diff
@@ -566,6 +570,62 @@ func (a *Properties) validateAddons() error {
 	return nil
 }
 
+func (a *Properties) validateCustomManifestData() error {
+	if a.OrchestratorProfile.KubernetesConfig != nil {
+		if a.OrchestratorProfile.KubernetesConfig.PodSecurityPolicyConfig["data"] != "" {
```

Comment:
A lot of duplication here. We can create a private method that performs the gzip and B64 validation and call that multiple times.

Reply:
Can we also add uni tests?

Reply:
thanks @tariq1890.  I couldn't get B64 data to work with either the custom addon scripts or manifests. Perhaps I am missing something.

Yes I can create a private method and add unit tests.

Reply:
Could you let us know the error you got when you tried custom add-on scripts? Feel free to open an issue for that as well.

Reply:
Yes, I will build a cluster with current master and raise an issue with the error.

Reply:
@tariq1890 I have raised an issue here: https://github.com/Azure/acs-engine/issues/4102

Reply:
On second thought. Let's do away with this validation. This is an experimental feature and we are yet to actually decide a formalized UX for this.